app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CategoriesController < ApplicationController
+
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def check_for_duplicates(category)
+    if Category.exists?(title: category.title)
+      flash[:error] = "#{category.title} already exists."
+    end
+  end
+
+  def create
+    @category = Category.new(category_params)
+    check_for_duplicates(@category)
```

Comment:
this is a good idea - but when you call `@category.save` on the next line, it will go ahead and check to see if the category exists (as long as you have the model validation)